### PR TITLE
Reregister node when periodic fingerprint changes node properties

### PR DESCRIPTION
**Pull request by iverberk**

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -658,6 +656,39 @@ func (c *Client) run() {
 	}
 }
 
+func (c *Client) hasNodeChanged(oldAttrHash uint64, oldMetaHash uint64) (bool, uint64, uint64) {
```
Comment (dadgar) [inline client/client.go]: Put a comment on what this does/its purpose and its return values.

File: client/client.go [hunk 2]
```diff
@@ -658,6 +656,39 @@ func (c *Client) run() {
...
+		c.logger.Printf("[DEBUG] client: unable to calculate node attributes hash: %v", err)
+	}
+	// Calculate node meta map hash
+	newMetaHash, err := hashstructure.Hash(c.config.Node.Meta, nil)
+	if err != nil {
+		c.logger.Printf("[DEBUG] client: unable to calculate node meta hash: %v", err)
+	}
+	if newAttrHash != oldAttrHash || newMetaHash != oldMetaHash {
+		return true, newAttrHash, newMetaHash
+	} else {
```
Comment (dadgar) [inline client/client.go]: Can remove the else block and just keep the return. Common Go style

**Comment by dadgar** [comment]
LGTM!

File: client/client.go (hunk 1 above)
Comment (iverberk) [inline client/client.go]: Comments added.

File: client/client.go (hunk 2 above)
Comment (iverberk) [inline client/client.go]: Done.

**Pull request merged**